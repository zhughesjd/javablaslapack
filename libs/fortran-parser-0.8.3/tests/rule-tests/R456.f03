
! Testing type_param_spec_list. Courtesy of Craig.
type(truth(5)) :: beauty
type(ugly(dim=a)) :: lies
type(this(3,4,that=5)) :: theother
end 
