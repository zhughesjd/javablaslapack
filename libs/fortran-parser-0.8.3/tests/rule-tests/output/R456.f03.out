********************************************
args[1]: R456.f03
R1101:main-program__begin:
R 501:type-declaration-stmt__begin:
R 456:type-param-spec-list__begin:
R 406:int-literal-constant: digitString=[@7,63:63='5',<10>,3:11] kindParam=null
R 402:type-param-value: hasExpr=true hasAsterisk=false hasColon=false
R 456:type-param-spec: keyWord=null
R 456:type-param-spec-list: count=1
R 455:derived-type-spec: typeName=[@5,57:61='truth',<206>,3:5] hasTypeParamSpecList=true
R 502:declaration-type-spec: type=501
R 504:entity-decl-list__begin:
R 504:entity-decl: id=[@13,70:75='beauty',<206>,3:18]
R 504:entity-decl-list: count=1
R 501:type-declaration-stmt: numAttributes=0
R 501:type-declaration-stmt__begin:
R 456:type-param-spec-list__begin:
R 613:part-ref: id=[@21,91:91='a',<206>,4:14] hasSelectionSubscriptList=false hasImageSelector=false
R 612:data-ref: numPartRef=1
R 402:type-param-value: hasExpr=true hasAsterisk=false hasColon=false
R 456:type-param-spec: keyWord=[@19,87:89='dim',<206>,4:10]
R 456:type-param-spec-list: count=1
R 455:derived-type-spec: typeName=[@17,82:85='ugly',<206>,4:5] hasTypeParamSpecList=true
R 502:declaration-type-spec: type=501
R 504:entity-decl-list__begin:
R 504:entity-decl: id=[@27,98:101='lies',<206>,4:21]
R 504:entity-decl-list: count=1
R 501:type-declaration-stmt: numAttributes=0
R 501:type-declaration-stmt__begin:
R 456:type-param-spec-list__begin:
R 406:int-literal-constant: digitString=[@33,113:113='3',<10>,5:10] kindParam=null
R 402:type-param-value: hasExpr=true hasAsterisk=false hasColon=false
R 456:type-param-spec: keyWord=null
R 406:int-literal-constant: digitString=[@35,115:115='4',<10>,5:12] kindParam=null
R 402:type-param-value: hasExpr=true hasAsterisk=false hasColon=false
R 456:type-param-spec: keyWord=null
R 406:int-literal-constant: digitString=[@39,122:122='5',<10>,5:19] kindParam=null
R 402:type-param-value: hasExpr=true hasAsterisk=false hasColon=false
R 456:type-param-spec: keyWord=[@37,117:120='that',<206>,5:14]
R 456:type-param-spec-list: count=3
R 455:derived-type-spec: typeName=[@31,108:111='this',<206>,5:5] hasTypeParamSpecList=true
R 502:declaration-type-spec: type=501
R 504:entity-decl-list__begin:
R 504:entity-decl: id=[@45,129:136='theother',<206>,5:26]
R 504:entity-decl-list: count=1
R 501:type-declaration-stmt: numAttributes=0
R 204:specification-part: numUseStmts=0 numImportStmts=0 numDeclConstructs=3
R9999:end-of-stmt:
R1103:end-program-stmt: id=null
R1101:main-program: hasProgramStmt=false hasExecutionPart=false hasInternalSubprogramPart=false
Parser exiting normally
********************************************
